plugins/modules/jdiff.py

```diff
@@ -0,0 +1,228 @@
+                  "ipv4": {
+                      "accepted_prefixes": 90,
+                      "sent_prefixes": 0
+                  }
+              }
+          }
+      }
+
+- name: "PARAMETER - VALIDATE PEER TYPE"
+  jdiff:
+    check_type: "parameter_match"
+    evaluate_args:
+      value_to_compare: "{{ data_to_compare }}"
+      mode: match
+      params:
+        linkType: external
+    jmespath: peers[*].[$ip$,linkType]
+  register: result
+  vars:
+    data_to_compare: |
+      {
+          "peers": [
+              {
+                  "ip": "10.1.0.0",
+                  "linkType": "external"
+              },
+              {
+                  "ip": "10.2.0.0",
+                  "linkType": "external"
+              }
+          ]
+      }
+
+- name: "REGEX - VALIDATE MAC FORMAT"
+  jdiff:
+    check_type: "regex"
+    evaluate_args:
+      value_to_compare: "{{ data_to_compare }}"
+      regex: "^([0-9a-fA-F]{2}(:|-)){5}([0-9a-fA-F]{2})$"
+      mode: match
+    jmespath: interfaces.*.[$name$,burnedInAddress]
+  register: result
+  vars:
+    data_to_compare: |
+      {
+          "interfaces": {
+              "Management1": {
+                  "burnedInAddress": "08:00:27:e6:b2:f8",
+                  "name": "Management1"
+              },
+              "Management2": {
+                  "burnedInAddress": "08-00-27-e6-b2-f9",
+                  "name": "Management2"
+              }
+          }
+      }
+
+- name: "OPERATOR - VALIDATE RX LEVEL WITHIN RANGE"
+  jdiff:
+    check_type: "operator"
+    evaluate_args:
+      value_to_compare: "{{ data_to_compare }}"
+      params:
+        params:
+          mode: in-range
+          operator_data: [-8, -2]
+    jmespath: ports[*].[$name$,RxPower]
+  register: result
+  vars:
+    data_to_compare: |
+      {
+          "ports": [
+              {
+                  "name": "1/1",
+                  "RxPower": -3.83,
+                  "state": "Connected"
+              },
+              {
+                  "name": "1/2",
+                  "RxPower": -3.21,
+                  "state": "Connected"
+              }
+          ]
+      }
+"""
+
+RETURN = """
+changed:
+    description: Indicates if change was made - always False.
+    returned: success
+    type: bool
+fail_details:
+    description: deepdiff output indicating where the check failed
+    returned: success
+    type: dict
+success:
+    description: Indicates if the check was successful.
+    returned: success
+    type: bool
+"""
+
+from ansible.module_utils.basic import AnsibleModule
+
+
+def main():
+    """Module function."""
+    argument_spec = dict(
+        check_type=dict(type='str', required=True),
+        evaluate_args=dict(type='dict', required=True),
+        jmespath=dict(type='str', default='*'),
+        exclude=dict(type='list', elements='str', default=[]),
+    )
+
+    AnsibleModule(
+        argument_spec=argument_spec,
+        required_together=[['check_type', 'evaluate_args']],
```

Comment:
if it's already required alone, why together?

Reply:
When calling the module, you need to specify `check_type` to call factory class together with `evaluate_args` which are being passed to `.evaluate()` method.

https://docs.ansible.com/ansible/latest/dev_guide/developing_program_flow_modules.html
```
If at least one of these options are specified, the other ones from the same sequence must all be present.
```